src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectToolsCommandResolver.cs

```diff
@@ -189,19 +192,33 @@ private CommandSpec ResolveFromProjectTools(CommandResolverArguments commandReso
             return commandSpec;
         }
 
-        private LockFile GetToolLockFile(
+        private IEnumerable<string> GetPossiblePackageRoots(IProject project)
+        {
+            if (project.TryGetLockFile(out LockFile lockFile))
```

Comment:
I believe we should fail if it can't find a project.assets.json for the project that we are trying to run the tool from.